internal/app/init/internal/rootfs/rootfs.go

```diff
@@ -40,27 +40,37 @@ func ip() string {
 
 // Prepare creates the files required by the installed binaries and libraries.
 // nolint: gocyclo
-func Prepare(s string, data *userdata.UserData) (err error) {
-	// Enable IP forwarding.
-	if err = proc.WriteSystemProperty(&proc.SystemProperty{Key: "net.ipv4.ip_forward", Value: "1"}); err != nil {
-		return
-	}
-	// Kernel Self Protection Project recommended settings.
-	if err = kernelHardening(); err != nil {
-		return
-	}
-	// Create /etc/hosts.
-	hostname, err := os.Hostname()
-	if err != nil {
-		return
-	}
-	ip := ip()
-	if err = etc.Hosts(s, hostname, ip); err != nil {
-		return
-	}
-	// Create /etc/resolv.conf.
-	if err = etc.ResolvConf(s); err != nil {
-		return
+func Prepare(s string, inContainer bool, data *userdata.UserData) (err error) {
+	if !inContainer {
+		// Enable IP forwarding.
+		if err = proc.WriteSystemProperty(&proc.SystemProperty{Key: "net.ipv4.ip_forward", Value: "1"}); err != nil {
+			return
+		}
+		// Kernel Self Protection Project recommended settings.
+		if err = kernelHardening(); err != nil {
+			return
+		}
+		// Create /etc/hosts.
+		var hostname string
+		if hostname, err = os.Hostname(); err != nil {
+			return
+		}
+		ip := ip()
+		if err = etc.Hosts(s, hostname, ip); err != nil {
+			return
+		}
+		// Create /etc/resolv.conf.
+		if err = etc.ResolvConf(s); err != nil {
+			return
+		}
+	} else {
+		// See https://github.com/coredns/coredns/tree/master/plugin/loop#troubleshooting
+		const resolvConf = `nameserver 8.8.4.4
```

Comment:
nit, should this really be a const if we're just making use of it immediately below?

Reply:
yeah I should probably move this out of the func.